test/chainweb/p2p/block_hash_test.exs

```diff
@@ -173,4 +284,108 @@ defmodule Kadena.Chainweb.P2P.BlockHashTest do
     {:error, %Error{status: :network_error, title: :nxdomain}} =
       BlockHash.retrieve(location: "col1", network_id: :mainnet01)
   end
+
+  describe "retrieve_branches/2" do
+    setup do
+      Application.put_env(:kadena, :http_client_impl, CannedBlockHashRequests)
+
+      on_exit(fn ->
+        Application.delete_env(:kadena, :http_client_impl)
+      end)
+
+      lower = ["4kaI5Wk-t3mvNZoBmVECbk_xge5SujrVh1s8S-GESKI"]
+      upper = ["HHEJ8CfvcweMTfvSMBYlXLWv0v25Mt-4bK3RUi_L6ls"]
+
+      success_response =
+        {:ok,
+         %BlockHashResponse{
+           items: [
+             "HHEJ8CfvcweMTfvSMBYlXLWv0v25Mt-4bK3RUi_L6ls",
+             "gmV-pRi50fUcy2i9v8cba_HDjw2_GP47RKgpKD-0av8",
+             "jVP-BDWC93RfDzBVQxolPJi7RcX09ax1IMg0_I_MNIk"
+           ],
+           limit: 3,
+           next: nil
+         }}
+
+      success_response2 =
+        {:ok,
+         %BlockHashResponse{
+           items: [
+             "HHEJ8CfvcweMTfvSMBYlXLWv0v25Mt-4bK3RUi_L6ls",
+             "gmV-pRi50fUcy2i9v8cba_HDjw2_GP47RKgpKD-0av8",
+             "jVP-BDWC93RfDzBVQxolPJi7RcX09ax1IMg0_I_MNIk",
+             "4kaI5Wk-t3mvNZoBmVECbk_xge5SujrVh1s8S-GESKI",
+             "M4doD-jMHyxi4TvfBDUy3x9VMkcLxgnpjtvbbd0yUQA",
+             "3eH11vI_wZuP3lEKcilfCx89_kZ78nFuJJbty44iNBo",
+             "r21zg8E011awAbEghzNBOI4RtKUZ-wHLkUwio-5dKpE"
+           ],
+           limit: 7,
+           next: nil
+         }}
+
+      success_response3 =
+        {:ok,
+         %BlockHashResponse{
+           items: [],
+           limit: 0,
+           next: nil
+         }}
+
+      %{
+        lower: lower,
+        upper: upper,
+        success_response: success_response,
+        success_response2: success_response2,
+        success_response3: success_response3
```

Comment:
all three cases have the `next` property set to zero, we should have a test using the `next` property?